RetakesAllocatorCore/Menus/WeaponsMenu.cs

```diff
@@ -0,0 +1,240 @@
+        
+        player.PrintToChat($"{MessagePrefix} You selected {weaponName} as CT Primary!");
+        HandleAllocation(player, CsTeam.CounterTerrorist, weaponName);
+
+        OpenCtSecondaryMenu(player);
+    }
+
+    private void OpenCtSecondaryMenu(CCSPlayerController player)
+    {
+        var menu = new ChatMenu($"{MessagePrefix} Select a CT Secondary Weapon");
+
+        foreach (var weapon in WeaponHelpers.GetPossibleWeaponsForRoundType(RoundType.Pistol, CsTeam.Terrorist))
+        {
+            menu.AddMenuOption(weapon.ToString(), OnCTSecondarySelect);
+        }
+
+        menu.AddMenuOption("Exit", OnSelectExit);
+
+        ChatMenus.OpenMenu(player, menu);
+        CreateMenuTimeoutTimer(player);
+    }
+
+    private void OnCTSecondarySelect(CCSPlayerController player, ChatMenuOption option)
+    {
+        if (!PlayersInGunsMenu.Contains(player))
+        {
+            return;
+        }
+
+        var weaponName = option.Text;
+        
+        // TODO: Separate allocation for CT pistol and T pistol
+        player.PrintToChat($"{MessagePrefix} You selected {weaponName} as CT Secondary!");
+        HandleAllocation(player, CsTeam.CounterTerrorist, weaponName);
+
+        // OpenGiveAwpMenu(player);
+        OnMenuComplete(player);
+    }
+
+    // private void OpenGiveAwpMenu(CCSPlayerController player)
+    // {
+    //     var menu = new ChatMenu($"{MessagePrefix} Select when to give the AWP");
+    //
+    //     menu.AddMenuOption("Never", OnGiveAwpSelect);
+    //     menu.AddMenuOption("Sometimes", OnGiveAwpSelect);
+    //     menu.AddMenuOption("Always", OnGiveAwpSelect);
+    //
+    //     menu.AddMenuOption("Exit", OnSelectExit);
+    //
+    //     ChatMenus.OpenMenu(player, menu);
+    // }
+    //
+    // private void OnGiveAwpSelect(CCSPlayerController player, ChatMenuOption option)
+    // {
+    //     if (!PlayersInGunsMenu.Contains(player))
+    //     {
+    //         return;
+    //     }
+    //
+    //     player.PrintToChat($"{MessagePrefix} You selected {option.Text} as when to give the AWP!");
+    //
+    //     switch (option.Text)
+    //     {
+    //         case "Never":
+    //             TODO: Implement weapon allocation selection
+    //             break;
+    //         case "Sometimes":
+    //             TODO: Implement weapon allocation selection
+    //             break;
+    //         case "Always":
+    //             TODO: Implement weapon allocation selection
+    //             break;
+    //     }
+    //
+    //     player.PrintToChat($"{MessagePrefix} You have finished setting up your weapons!");
+    //     player.PrintToChat($"{MessagePrefix} The weapons you have selected will be given to you at the start of the next round!");
+    //
+    //     PlayersInGunsMenu.Remove(player);
+    // }
+
+    private static void HandleAllocation(CCSPlayerController player, CsTeam team, string weapon)
+    {
+        var response = OnWeaponCommandHelper.Handle(
```

Comment:
I am using a similar approach to handle preference setting here https://github.com/yonilerner/cs2-retakes-allocator/pull/40/files#diff-9a978eb8236c5e3558e27805427517a303f95b12092f4d72a21ec7e69a57208dR293

seems legit. Might need to rename this at some point, but i think in a separate PR so we don't get merge conflicts for nothing

Reply:
Yeah it has some other logic bundled in with it but it was actually pretty useful